cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/repo.go

```diff
@@ -252,7 +254,19 @@ func (s *Server) newRepo(ctx context.Context, request *corev1.AddPackageReposito
 	passCredentials := auth != nil && auth.PassCredentials
 	interval := request.GetInterval()
 
-	if fluxRepo, err := newFluxHelmRepo(name, typ, url, interval, secret, passCredentials); err != nil {
+	// Get Flux-specific values
+	provider := ""
+	var customDetail *v1alpha1.FluxPackageRepositoryCustomDetail
+	if request.CustomDetail != nil {
+		customDetail = &v1alpha1.FluxPackageRepositoryCustomDetail{}
+		if err := request.CustomDetail.UnmarshalTo(customDetail); err != nil {
+			return nil, status.Errorf(codes.InvalidArgument, "customDetail could not be parsed due to: %v", err)
+		}
+		log.Infof("fluxv2 customDetail: [%v]", customDetail)
+		provider = customDetail.Provider
```

Comment:
Shouldn't we fail at this point if there is a provider and repo type is not `HelmRepositoryTypeOCI`?

Reply:
again, I prefer to let flux do this validation, which they already do. See the note in https://fluxcd.io/flux/components/source/helmrepositories/#provider